src/Data/Widget/Adapters/Adapter.elm

```diff
@@ -1,11 +1,56 @@
-module Data.Widget.Adapters.Adapter exposing (Adapter(..))
+module Data.Widget.Adapters.Adapter exposing (Adapter(..), decoder)
 
--- TODO these are hardcoded for now but will likely have to be a set of
--- defaults plus optional user defined adapters
--- e.g. "use row 3 for headings & row 4-6 for data"
+import Data.Widget.Adapters.MetricAdapter as MetricAdapter exposing (Config, defaultConfig)
+import Json.Decode as Decode exposing (Decoder, maybe)
+import Json.Decode.Pipeline as Pipeline exposing (custom, decode, hardcoded, required, optional)
+import Data.Widget.Adapters.CellPosition exposing (CellPosition, cellPositionDecoder)
+
+
+type alias Definition =
+    { type_ : String
+    , config : Maybe Config
+    }
 
 
 type Adapter
     = TABLE
     | BAR_CHART
     | HEAT_MAP
+    | METRIC MetricAdapter.Config
+
+
+decoder : Decoder Adapter
+decoder =
+    definitionDecoder
+        |> Decode.andThen
+            (\definition ->
+                case definition.type_ of
+                    "TABLE" ->
+                        Decode.succeed TABLE
+
+                    "BAR_CHART" ->
+                        Decode.succeed BAR_CHART
+
+                    "HEAT_MAP" ->
+                        Decode.succeed HEAT_MAP
+
+                    "METRIC" ->
+                        Decode.succeed <| METRIC (definition.config |> Maybe.withDefault MetricAdapter.defaultConfig)
+
+                    somethingElse ->
+                        Decode.fail <| "Unknown adapter: " ++ somethingElse
+            )
+
+
+definitionDecoder : Decoder Definition
+definitionDecoder =
+    decode Definition
+        |> required "type_" Decode.string
+        |> optional "config" (maybe adapterConfigDecoder) Nothing
+
+
```

Comment:
@msp do you think that `adapterConfigDecoder` belongs in the `MetricsAdapter` module?